Face/FindSimilar/Program.cs

```diff
@@ -0,0 +1,198 @@
+                    return;
+                }
+
+                Console.WriteLine($"{detectedFaces.Count} faces detected from image `{sourceImageFileName}`.");
+                if (detectedFaces[0].FaceId == null)
+                {
+                    Console.WriteLine("[Error] Parameter `returnFaceId` of `DetectWithStreamAsync` must be set to `true` (by default) for verification purpose.");
+                    return;
+                }
+
+                // Find similar example for find similar faces from the face list with similar confidence.
+                var similarResults = Client.Face.FindSimilarAsync(detectedFaces[0].FaceId.Value, faceListId).Result;
+                foreach (var similarResult in similarResults)
+                {
+                    Console.WriteLine($"Faces from {sourceImageFileName} & {persistedFaces.Find(p => p.PersistedFaceId == similarResult.PersistedFaceId).UserData} are of the same (Positive) person, similarity confidence: {similarResult.Confidence}.");
+                }
+            }
+
+            // Delete the face list.
+            await Client.FaceList.DeleteAsync(faceListId);
+            Console.WriteLine($"{Environment.NewLine}Delete FaceList {faceListId}");
+
+            Console.WriteLine("\nPress ENTER to exit.");
+        }
+
+        public static async void FindSimilar_in_LargeFaceList()
+        {
+            List<string> targetImageFileNames =
+                new List<string> { "Family1-Dad1.jpg", "Family1-Dad2.jpg", "Family1-Son1.jpg" };
+            string sourceImageFileName = "Family1-Dad1.jpg";
+
+            // Create a large face list.
+            string largeFaceListId = Guid.NewGuid().ToString();
+            Console.WriteLine($"Create large face list {largeFaceListId} {Environment.NewLine}");
+            await Client.LargeFaceList.CreateAsync(largeFaceListId, largeFaceListId, "large face list for sample");
+
+            foreach (var targetImageFileName in targetImageFileNames)
+            {
+                // Read image file.
+                using (FileStream stream = new FileStream(Path.Combine("Images", targetImageFileName), FileMode.Open))
+                {
+                    // Add face to the large face list.
+                    Console.WriteLine($"add faces to large face list from image `{targetImageFileName}`.");
+                    var faces = Client.LargeFaceList.AddFaceFromStreamAsync(largeFaceListId, stream, targetImageFileName).Result;
+                    if (faces == null)
+                    {
+                        Console.WriteLine($"[Error] No persisted face from image `{targetImageFileName}`.");
+                        return;
+                    }
+                }
+            }
+
+            // Start to train the large face list.
+            Console.WriteLine($"Train large face list {largeFaceListId} {Environment.NewLine}");
+            await Client.LargeFaceList.TrainAsync(largeFaceListId);
```

Comment:
We need the logic of waiting for the success of training. Reference: https://github.com/Microsoft/Cognitive-Face-Windows/blob/master/Sample-WPF/Controls/FaceIdentificationPage.xaml.cs#L407-L416

Reply:
Same for (large) person group in the identification sample.

Reply:
done